packages/workspace/src/browser/workspace-service.ts

```diff
@@ -81,84 +75,125 @@ export class WorkspaceService implements FrontendApplicationContribution {
     tryGetRoots(): FileStat[] {
         return this._roots;
     }
+    isMetaDataSaved(): boolean {
+        return !!this._workspaceMetaDataFile;
+    }
 
     protected readonly onWorkspaceChangeEmitter = new Emitter<FileStat[]>();
     get onWorkspaceChanged(): Event<FileStat[]> {
         return this.onWorkspaceChangeEmitter.event;
     }
 
+    protected handleWorkspaceMetadataChange(): void {
+        if (this._workspaceMetaDataFile) {
+            this.watcher.onFilesChanged(event => {
```

Comment:
you should not forget to unwatch old files, if a workspace file has been changed